task6_Shai_Axelrod/Task06/src/DataBase.cpp

```diff
@@ -0,0 +1,75 @@
+#include "DataBase.h"
+#include <iostream>
+#include <algorithm>
+#include <fstream>  
+#include <sstream>
+#include <string>
+#include <iomanip>
+
+void DataBase::fillFromFile(std::string fileName)
+{
+    std::ifstream inputFile(fileName);
+    std::string line;
+    std::string cityName;
+
+    while (std::getline(inputFile, line)) {
+        std::istringstream iss(line);
+
+        if (line.empty()) {
+            continue; // Skip empty lines
+        }
+        else if (iss >> std::quoted(cityName)) {
+            // Read city name
+            Location location;
+            std::string coordinates;
+
+            // Read coordinates as a single string
+            if (std::getline(inputFile, coordinates)) {
+                std::istringstream coordsStream(coordinates);
+
+                // Extract the coordinates using the '-' as a delimiter
+                std::string xStr, yStr;
+                std::getline(coordsStream, xStr, '-');
+                std::getline(coordsStream, yStr);
+
+                location.x = std::stod(xStr);
+                location.y = std::stod(yStr);
+                if (_data.find(line) == _data.end())
+                {                    
+                    _data[line] = location;
+                }
+                
+            }
+        }
+    }
+    if (_data.empty())
+    {
+        throw std::exception("no cities found in file");
+    }   
+}
+
+
+std::vector<std::pair<std::string, Location>> DataBase::getCloseCities(std::string cityName, float radius, DistanceFunction distFormula)
+{
+    Location cityLoc = getCityLoc(cityName);
+    std::vector<std::pair<std::string, Location>> filtered;
+    // retrieve the cities within the square created by the radius && within the radius
```

Comment:
you are missing 1 step in finding the cities in radius, you should find vertical and horizontal rectangles, and then to find the intersection square, and to find a circle inside the square.